map/src/manager/track/TracksManager.js

```diff
@@ -788,49 +560,14 @@ export function findGroupByName(groups, groupName) {
     return null;
 }
 
-// after success upload from Local to Cloud
-// download it and use as current Cloud track
-async function downloadAfterUpload(ctx, file) {
-    const createState = {
-        enable: false, // stop-editor
-    };
-
-    // cleanup
-    if (ctx.createTrack?.enable && ctx.selectedGpxFile) {
-        createState.closePrev = {
-            file: _.cloneDeep(ctx.selectedGpxFile),
-        };
-    }
-
-    ctx.setCreateTrack({ ...createState });
+export function validName(name) {
+    return name !== '' && name.trim().length > 0;
+}
 
-    const URL = `${process.env.REACT_APP_USER_API_SITE}/mapapi/download-file`;
-    const qs = `?type=${encodeURIComponent(file.type)}&name=${encodeURIComponent(file.name)}`;
-    const newGpxFiles = Object.assign({}, ctx.gpxFiles);
-    newGpxFiles[file.name] = {
-        url: URL + qs,
-        clienttimems: file.clienttimems,
-        updatetimems: file.updatetimems,
-        name: file.name,
-        type: 'GPX',
-    };
-    const f = await Utils.getFileData(newGpxFiles[file.name]);
-    const gpxfile = new File([f], file.name, {
-        type: 'text/plain',
-    });
-    const track = await TracksManager.getTrackData(gpxfile);
-    if (isEmptyTrack(track) === false) {
-        const type = OBJECT_TYPE_CLOUD_TRACK;
-        ctx.setUpdateInfoBlock(true);
-        ctx.setCurrentObjectType(type);
-        track.name = file.name;
-        Object.keys(track).forEach((t) => {
-            newGpxFiles[file.name][`${t}`] = track[t];
-        });
-        newGpxFiles[file.name].analysis = TracksManager.prepareAnalysis(newGpxFiles[file.name].analysis);
-        ctx.setGpxFiles(newGpxFiles);
-        ctx.setSelectedGpxFile(Object.assign({}, newGpxFiles[file.name]));
-    }
+export function hasExistTrack(name, folder, tracks) {
+    const folderName = folder.title ? folder.title : folder;
+    const foundFolder = findGroupByName(tracks, folderName);
+    return foundFolder ? foundFolder.files.find((f) => TracksManager.prepareName(f.name) === name) : false;
```

Comment:
1) maybe better name would be folderHasTrack() or isTrackExists()
2) as this is boolean function, better to use "some()" not "find()"
3) please replace find->some in the parent (for local tracks)